src/apps/backend/modules/otp/types.ts

```diff
@@ -0,0 +1,61 @@
+import { PhoneNumber } from '../account/types';
+import { ApplicationError } from '../application';
+import { HttpStatusCodes } from '../http';
+
+export enum OtpStatus {
+  EXPIRED = 'EXPIRED',
+  PENDING = 'PENDING',
+  SUCCESS = 'SUCCESS',
+}
+
+export class Otp {
+  id: string;
+  otpCode: string;
+  phoneNumber: PhoneNumber;
+  status: OtpStatus;
+}
+
+export enum OtpErrorCode {
+  INCORRECT_OTP = 'OTP_ERR_01',
+  OTP_EXPIRED = 'OTP_ERR_02',
+  FAILED_TO_CREATE_OTP = 'OTP_ERR_03',
+}
+
+export type CreateOTPParams = {
+  phoneNumber: PhoneNumber;
+};
+
+export type VerifyOTPParams = {
+  otpCode: string;
+  phoneNumber: PhoneNumber;
+};
+
+export class OtpIncorrectError extends ApplicationError {
+  code: OtpErrorCode;
+
+  constructor() {
+    super('Please provide the correct OTP to login.');
+    this.code = OtpErrorCode.INCORRECT_OTP;
+    this.httpStatusCode = HttpStatusCodes.UNAUTHORIZED;
```

Comment:
I checked it on various resources, it is suggested to use 401 which is unauthorized. Error code for wrong credentials.

Reply:
So say you are trying to access a resource (say a task of User 1, but you are User 2), system can return - you are not authorized to access the resource. 

But when I sending a wrong OTP, system cannot tell me I am not authorized to perform that operation, ofcourse I am.

Reply:
If we provide the wrong access token or the access token is expired, we have the status as Unauthorized.
Because we require it to access the authorized resources.
We have also done the same for the invalid password.
Shouldn't it be the same for the OTP?

Reply:
@jjalan Sir, should I go with HttpStatusCodes.BadRequest or HttpStatusCodes.Unauthorized for this?

Reply:
Sounds like invalid password code should be updated to bad request too.